File: C Tasks/auto.c
/*The Person class represents the owner of the car and has:*/

/*firstName*/
/*lastName*/
/*age*/
/*The Auto class represents a car and has:*/

/*owner (Person)*/
/*maxSpeed*/
/*brand*/
/*name*/
/*price*/
/*Define toString method to both classes and print them.*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct Person {
    char firstName[10];
    char lastName[10];
    int age;
} person;

typedef struct Auto {
    person owner;
    int maxSpeed;
    char brand[10];
    char name[10];
} autostr;

void printPerson(person);
void printAuto(autostr);

main()
{
    char firstName[10];
    char lastName[10];
    int age;
    int maxSpeed;
    char brand[10];
    char name[10];
    
    printf("Enter your data: ");
    scanf("%9s %9s %d", firstName, lastName, &age);
    
    printf("Enter auto data: ");
    scanf("%9s %9s %d", brand, name, &maxSpeed);
    
    person rumen;
    autostr Ford;
    
    strcpy(rumen.firstName, firstName);
    strcpy(rumen.lastName, lastName);
    rumen.age = age;
    
    strcpy(Ford.brand, brand);
    strcpy(Ford.name, name);
    Ford.maxSpeed = maxSpeed;
    Ford.owner = rumen;

    printPerson(rumen);
    printAuto(Ford);
}

void printPerson(person person)
{
    printf("%s\t%s\t%d\n", person.firstName, person.lastName, person.age);
}

void printAuto(autostr Auto)
{
    printPerson(Auto.owner);
    printf("%d\t%s\t%s\n", Auto.maxSpeed, Auto.brand, Auto.name);
}

File: workspace1/week2/src/com/hackbulgaria/programming51/week2/Auto.java
//3-Auto
//In a com.hackbulgaria.programming51.week2 package create an Auto class, and a Person
//
//The Person class represents the owner of the car and has:
//
//firstName
//lastName
//age
//The Auto class represents a car and has:
//
//owner (Person)
//maxSpeed
//brand
//name
//price
//Define toString method to both classes and print them.
//
//Example: In main:
//
//Person ivan = new Person();
//ivan.firstName = "Ivan";
//ivan.lastName = "Ivanov";
//ivan.age = 25;
//
//Auto audi = new Auto();
//
//audi.owner = ivan;
//audi.maxSpeed = 200;
//audi.brand = "Audi";
//audi.name = "Begachka";
//audi.price = 5000;
//
//System.out.println(audi);
//Should print:
//
//Audi Begachka, Max Speed: 200, Price: 5000, Owner: Ivan Ivanov age: 25

package com.hackbulgaria.programming51.week2;

public class Auto {
	private String owner;
	private int maxSpeed;
	private String brand;
	private String name;
	private String price;
	
	public void addOwner(String name) {
		owner = name;
	}
	
	public String toString() {
		return owner + " " + maxSpeed + " " + brand + " " + name + " " + price;
	}
}